src/main/java/org/ow2/proactive/catalog/rest/util/parser/CatalogObjectParserFactory.java

```diff
@@ -45,7 +45,7 @@ public static CatalogObjectParserFactory get() {
         return factory;
     }
 
-    public CatalogObjectParser getParser(String type) {
+    public CatalogObjectParserInterface getParser(String type) {
         if ("workflow".equals(type))
```

Comment:
I don't think because it can be anything, and if not found you use the default parser that does nothing